app/controllers/application_controller.rb

```diff
@@ -1,3 +1,11 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  def current_tenant
+    current_user.tenant
+  end
+
+  def admin_user
+    redirect_to(root_path) unless current_user&.role == User::ADMIN_ROLE
```

Comment:
also, try to avoid using redirects from a helpers

Reply:
this is an application_controller, or you mean that this method is a helper?

Reply:
rename this method to 

application_controller
def require_admin!
  redirect_to(root_path) unless current_user&.role == User::ADMIN_ROLE
end

users_controller
before_action :authenti.... (as it's now)
before_action :require_admin!